src/ComputeSharp.Core/Intrinsics/Hlsl.Void.g.cs

```diff
@@ -0,0 +1,308 @@
+﻿using ComputeSharp.Core.Intrinsics.Attributes;
+using ComputeSharp.Exceptions;
+
+namespace ComputeSharp;
+
+/// <inheritdoc cref="Hlsl"/>
+partial class Hlsl
+{
+    /// <summary>
+    /// Discards the current pixel if the specified value is less than zero.
+    /// </summary>
+    /// <param name="x">The specified value.</param>
+    /// <remarks>
+    /// <para>Use the clip HLSL intrinsic function to simulate clipping planes if each component of the x parameter represents the distance from a plane.</para>
+    /// <para>Also, use the clip function to test for alpha behavior, as shown in the following example:</para>
+    /// <code>
+    /// Hlsl.Clip(color.A &lt; 0.1f ? -1 : 1);
+    /// </code>
+    /// <para>This method is an intrinsic and can only be used within a shader on the GPU. Using it on the CPU is undefined behavior.</para>
+    /// </remarks>
+    [HlslIntrinsicName("clip")]
+    public static void Clip(float x) => throw new InvalidExecutionContextException($"{typeof(Hlsl)}.{nameof(Clip)}({typeof(float)})");
```

Comment:
Do you know how this actually works? If it doesn't return a value, and it doesn't modify the parameter, what does it mean by "discards the current pixel"? 🤔

Reply:
I was actually looking at this in the HLSL intrinsics docs yesterday and scratching my head over it

Reply:
I _think_ it just kills the dispatch for that individual pixel?
Like, similar to `abort`, but just discarding the pixel? 🤔